### PR TITLE
Simple fix for the output from the text-based circuit drawer

### LINKED ISSUE #1167: Text-based circuit drawer output is scrambled in notebook unless print is called.
### Informations

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python 
from qiskit import *
from qiskit.tools.visualization import circuit_drawer

q = QuantumRegister(5)
qc = QuantumCircuit(q)
qc.ccx(q[1], q[2], q[3])
qc.ccx(q[0], q[2], q[4])
qc.x(q[4])
qc.x(q[3])

circuit_drawer(qc, output='text')
```

'                             \nq1_0: |0>────────────■───────\n                     │       \nq1_1: |0>──■─────────┼───────\n           │         │       \nq1_2: |0>──■─────────■───────\n         ┌─┴─┐┌───┐  │       \nq1_3: |0>┤ X ├┤ X ├──┼───────\n         └───┘└───┘┌─┴─┐┌───┐\nq1_4: |0>──────────┤ X ├┤ X ├\n                   └───┘└───┘'

### Steps to reproduce the problem
The output is messed up unless calling `print(circuit_drawer(qc, output='text'))`.  Therefore it is not a drop in replacement for the other methods which work fine in a notebook.

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
fixes #1167 by outputing a simple class object.

### Details and comments
Tests will need to wait for #1154 to be merged.

### CODE REVIEW

File: qiskit/tools/visualization/_circuit_visualization.py [hunk 1]
```diff
@@ -291,7 +304,7 @@ def _text_circuit_drawer(circuit, filename=None,
         line_length (int): Optional. Sometimes, your console is too small of the drawing. Give me
                            you maximum line length your console supports.
     Returns:
-        String: The drawing in a loooong string.
+        CircuitString: A class holding the drawing string.
```
[the remark this answers is not in the document]
Comment (nonhermitian) [reply qiskit/tools/visualization/_circuit_visualization.py]: This is done.

File: qiskit/tools/visualization/_circuit_visualization.py [hunk 2]
```diff
@@ -277,6 +277,19 @@ def qx_color_scheme():
 # -----------------------------------------------------------------------------
 # _text_circuit_drawer
 # -----------------------------------------------------------------------------
+class CircuitString():
+    """A class for holding the text string
+    instance of a quantum circuit
```
[the remark this answers is not in the document]
Comment (nonhermitian) [reply qiskit/tools/visualization/_circuit_visualization.py]: I will just subclass it.

**Comment by 1ucian0** [comment]
Oh. I didnt realize the existance of this PR while I was working on https://github.com/Qiskit/qiskit-terra/pull/1181

**Comment by 1ucian0** [comment]
I suggest closing this in favor of #1181

**Comment by jaygambetta** [comment]
ok i let @nonhermitian @1ucian0 and @mtreinish to decide on the fix. But we should try to coordinate a bit better :-)

**Comment by 1ucian0** [comment]
This triggered a bigger conversation about the output type of circuit_drawer that is happening in #1190. Therefore, this is status: on hold'ed until #1190 is decided.

Also, this PR is competing with PR #1181. The discussion should happen there.

**Comment by 1ucian0** [comment]
Closing this PR in favor of #1181, after talking with @nonhermitian.

**Pull request closed without merging**